code/game/objects/items.dm

```diff
@@ -170,6 +173,9 @@ GLOBAL_DATUM_INIT(welding_sparks, /mutable_appearance, mutable_appearance('icons
 /obj/item/Destroy()
 	flags &= ~DROPDEL	//prevent reqdels
 	QDEL_NULL(hidden_uplink)
+	if(objective_item)
+		message_admins("[src] has been destroyed at ([x],[y],[z]) in the location [loc].")
+		log_game("[src] has been destroyed at ([x],[y],[z]) in the location [loc].")
```

Comment:
not really a fan of this being in `Destroy()` either, you should register a signal to these items somewhere else and do the admin logging there.

Reply:
This

Reply:
I'll update this to a signal then

Reply:
Should be updated